machine/v1beta1/types_azureprovider.go

```diff
@@ -227,18 +292,33 @@ type DiskSettings struct {
 	EphemeralStorageLocation string `json:"ephemeralStorageLocation,omitempty"`
 }
 
-// ManagedDiskParameters is the parameters of a managed disk.
+// ManagedDiskParameters is the parameters of a OSDisk managed disk.
 type ManagedDiskParameters struct {
-	// StorageAccountType is the storage account type to use. Possible values include "Standard_LRS" and "Premium_LRS".
+	// StorageAccountType is the storage account type to use.
+	// Possible values include "Standard_LRS", "Premium_LRS".
 	StorageAccountType string `json:"storageAccountType"`
 	// DiskEncryptionSet is the disk encryption set properties
 	// +optional
 	DiskEncryptionSet *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
 }
 
+// DataDiskManagedDiskParameters is the parameters of a DataDisk managed disk.
+type DataDiskManagedDiskParameters struct {
+	// StorageAccountType is the storage account type to use.
+	// Possible values include "Standard_LRS", "Premium_LRS" and "UltraSSD_LRS".
```

Comment:
generated validation please.

Reply:
Ok I've done so through an Enum and created a new type with the various const for it.